### PR TITLE
Rename XxxRespone->XxxInfo

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**

### CODE REVIEW

File: sdk/storage/inc/blobs/blob_client.hpp [hunk 1]
```diff
@@ -192,7 +181,7 @@ namespace Azure { namespace Storage { namespace Blobs {
      * parameters to execute this function.
      * @return A SetAccessTierResponse on successfully setting the tier.
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_client.hpp:182]: Not consistent with the actual return type.

File: sdk/storage/inc/blobs/blob_client.hpp [hunk 2]
```diff
@@ -220,7 +209,7 @@ namespace Azure { namespace Storage { namespace Blobs {
      * @param options Optional parameters to execute this function.
      * @return A AbortCopyBlobResponse on successfully aborting.
      */
-    Azure::Core::Response<AbortCopyBlobResponse> AbortCopyFromUri(
+    Azure::Core::Response<AbortCopyBlobInfo> AbortCopyFromUri(
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_client.hpp:212]: What about `AbortCopy` API? Are they having the same name?

File: sdk/storage/inc/blobs/blob_client.hpp [hunk 3]
```diff
@@ -283,7 +272,7 @@ namespace Azure { namespace Storage { namespace Blobs {
      * @param options Optional parameters to execute this function.
      * @return A DeleteBlobResponse on successfully deleting.
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_client.hpp:273]: ditto.

Comment (katmsft) [reply sdk/storage/inc/blobs/blob_client.hpp:273]: And others.

File: sdk/storage/inc/blobs/blob_client.hpp [hunk 4]
```diff
@@ -192,7 +181,7 @@ namespace Azure { namespace Storage { namespace Blobs {
      * parameters to execute this function.
      * @return A SetAccessTierResponse on successfully setting the tier.
      */
-    Azure::Core::Response<SetAccessTierResponse> SetAccessTier(
+    Azure::Core::Response<SetAccessTierInfo> SetAccessTier(
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_client.hpp:184]: Can we be absolutely sure that there won't be any container access tier in the future? If not, then there would be breaking change or `SetAccessTierInfo` and `SetContainerAccessTierInfo`.

File: sdk/storage/inc/blobs/blob_responses.hpp [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+#include "protocol/blob_rest_client.hpp"
+
+#include <limits>
+#include <string>
+#include <utility>
+
+namespace Azure { namespace Storage { namespace Blobs {
+
+  struct BlobDownloadInfo
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_responses.hpp:14]: `DeleteContainerInfo`
`DeleteBlobInfo`
`UndeleteBlobInfo`
Those are all `<Verb><Object>Info`
But `BlobDownloadInfo` is `<Object><Verb>Info`.

File: sdk/storage/inc/blobs/blob_responses.hpp [hunk 6]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "protocol/blob_rest_client.hpp"
+
+#include <limits>
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_responses.hpp:8]: What's it for?

File: sdk/storage/inc/blobs/blob_responses.hpp [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+#include <string>
+#include <utility>
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_responses.hpp:10]: What's it for?

File: sdk/storage/inc/blobs/blob_responses.hpp [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+  struct BlobDownloadInfo
+  {
+    std::string ETag;
+    std::string LastModified;
+    int64_t ContentLength = 0;
+    BlobHttpHeaders HttpHeaders;
+    std::map<std::string, std::string> Metadata;
+    Blobs::BlobType BlobType = Blobs::BlobType::Unknown;
+    Azure::Core::Nullable<bool> ServerEncrypted;
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_responses.hpp:22]: Why is it `Nullable`?

File: sdk/storage/inc/blobs/blob_responses.hpp [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+  struct PageRange
+  {
+    int64_t Offset;
+    int64_t Length;
+  };
+
+  struct PageRangesInfo
+  {
+    std::string RequestId;
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_responses.hpp:34]: Do we still need to have the fabulous 4 here?

File: sdk/storage/inc/blobs/blob_client.hpp (hunk 4 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp:184]: Why would a container have access tier? It doesn't make any sense.

File: sdk/storage/inc/blobs/blob_responses.hpp (hunk 5 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_responses.hpp:14]: align with .Net SDK

File: sdk/storage/inc/blobs/blob_responses.hpp (hunk 8 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_responses.hpp:22]: I cannot remember. But I changed it from non-nullable to nullable. There must be an API that doesn't return this value in response headers.

File: sdk/storage/inc/blobs/blob_client.hpp (hunk 1 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp]: fixed

File: sdk/storage/inc/blobs/blob_client.hpp (hunk 2 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp:212]: There isn't such an API

File: sdk/storage/inc/blobs/blob_client.hpp (hunk 3 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp]: fixed

File: sdk/storage/inc/blobs/blob_responses.hpp (hunk 7 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_responses.hpp]: fixed

File: sdk/storage/inc/blobs/blob_responses.hpp (hunk 9 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_responses.hpp]: fixed

File: sdk/storage/inc/blobs/blob_client.hpp (hunk 4 above)
Comment (katmsft) [reply sdk/storage/inc/blobs/blob_client.hpp:184]: There might be the case that customer wants to manage the access tier at the container level. I'm not sure if that's going to be added in the future.
What I want to point out is that `SetAccessTierInfo` does not limit the scope to be `Blob`, then any addition to the scope under the `Azure::Storage::Blobs` namespace would be breaking change in the future. If this is considered not possible in your mind, I'm fine with the current naming.

File: sdk/storage/inc/blobs/blob_responses.hpp (hunk 5 above)
Comment (katmsft) [reply sdk/storage/inc/blobs/blob_responses.hpp:14]: Then I don't think .NET SDK is doing the correct thing here. @JeffreyRichter what do you think? Is it going to be challenged by the review board in the future?

File: sdk/storage/inc/blobs/blob_responses.hpp (hunk 6 above)
Comment (katmsft) [reply sdk/storage/inc/blobs/blob_responses.hpp]: fixed.

File: sdk/storage/inc/blobs/blob_responses.hpp (hunk 8 above)
Comment (katmsft) [reply sdk/storage/inc/blobs/blob_responses.hpp:22]: Maybe `GetBlob` REST with content range does not return this. Please give it a try when you have time. Not priority.

File: sdk/storage/inc/blobs/blob_client.hpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/inc/blobs/blob_client.hpp:212]: https://docs.microsoft.com/en-us/rest/api/storageservices/abort-copy-blob

Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp:212]: There's only one Copy and one Abort Copy API in client API. This aligns with other sdk

File: sdk/storage/inc/blobs/blob_responses.hpp (hunk 5 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_responses.hpp:14]: I think it's fine.

File: sdk/storage/inc/blobs/blob_client.hpp (hunk 4 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp]: fixed

**Pull request merged**